client/src/product-components/Overview.jsx

```diff
@@ -41,12 +42,25 @@ class Overview extends React.Component {
       // console.log(sizeArr[i], sizeObj[sizeArr[i]], '===========mark');
       var obj =  sizeObj[sizeArr[i]];
       obj['sku'] =  sizeArr[i];
-      sizeList.push(obj);
+      if (obj.quantity !== 0) {
+        sizeList.push(obj);
+      }
     }
     // console.log(sizeObj, sizeList, '=============sizeList');
     return sizeList;
   }
 
+  getSizeStock(sizeObj) {
+    var stock = 0;
+    for (var i in sizeObj) {
+      stock += sizeObj[i].quantity;
+    }
+    if (stock <= 0) {
```

Comment:
This is a rare case where a ternary really would be best

stock <= 0 ? true : false;